mlflow/utils/model_utils.py

```diff
@@ -162,6 +164,66 @@ def _validate_and_copy_code_paths(code_paths, path, default_subpath="code"):
     return code_dir_subpath
 
 
+def _infer_and_copy_code_paths(flavor, path, default_subpath="code"):
+    # Capture all imported modules with full module name during loading model.
+    modules = _capture_imported_modules(path, flavor, record_full_module=True)
+
+    all_modules = set(modules)
+
+    def get_parent_module(_module):
+        return _module[0: _module.rindex(".")]
```

Comment:
This can be pulled out. We don't need to define this function every time `_infer_and_copy_code_paths` is called.

Reply:
Inner functions are error-prone.